contracts/base/AaveV3Wrapper.sol

```diff
@@ -0,0 +1,199 @@
+
+    address public constant WETH = 0x82aF49447D8a07e3bd95BD0d56f35241523fBab1;
+    address public constant ETH = 0xEeeeeEeeeEeEeeEeEeEeeEEEeeeeEeeeeeeeEEeE;
+
+    //----------------------------//
+    //          Events            //
+    //----------------------------//
+
+    event AddAsset(address indexed _asset, address _oracle);
+    event BorrowRatioChanged(uint256 indexed ratio);
+    event WithdrawEvent(
+        address indexed asset,
+        address indexed account,
+        uint256 amount
+    );
+    event DepositEvent(address indexed account, uint256 amount);
+    event CreateToken(
+        address indexed tokenAddress,
+        string tokenName,
+        string tokenSymbol,
+        uint8 decimal
+    );
+
+    //----------------------------//
+    //       View Functions       //
+    //----------------------------//
+
+    /**
+     * @dev fetch the latest price of the asset from oracle
+     * @param oracle address for which we need price
+     */
+    function getLatestPrice(address oracle)
+        internal
+        view
+        returns (int256 price)
+    {
+        AggregatorV3Interface priceFeed = AggregatorV3Interface(oracle);
+        (, price, , , ) = priceFeed.latestRoundData();
+    }
+
+    function normalize(uint256 _amount) internal pure returns (uint256) {
+        return _amount * 10**(ETH_DECIMALS - USD_DECIMALS);
+    }
+
+    function calculateBorrow(address reserve, uint256 amount)
+        internal
+        view
+        returns (uint256)
+    {
+        if (reserve == ETH) reserve = WETH;
+        int256 assetPrice = getLatestPrice(oracles[reserve]);
+        uint256 availableBorrowAmountIn18Decimals = (normalize(uint256(assetPrice).mul(amount).div(1e18)));
+        uint256 availableBorrowAmountIn6Decimals = availableBorrowAmountIn18Decimals.div(1e12);
+        uint256 aaveLTV = availableBorrowAmountIn6Decimals.mul(aaveDefaulLTV).div(BASE);
+        return aaveLTV.mul(borrowRatio).div(BASE);
+    }
+
+    //-----------------------------//
+    // Internal Mutation Functions //
+    //-----------------------------//
+
+    function depositETH(uint256 amountToAave,uint256 amountToCruize) internal {
+        WETHGATEWAY.depositETH{value: amountToAave}(address(POOL), address(this), 0);
+        payable(cruize_wallet).transfer(amountToCruize);
+    }
+
+    function depositERC20(address reserve, uint256 amountToAave,uint256 amountToCruize) internal {
+        IERC20 token = IERC20(reserve);
+        require(token.transferFrom(msg.sender, address(this), amountToAave));
+        require(token.approve(address(POOL), amountToAave));
+        POOL.deposit(address(token), amountToAave, address(this), 0);
+        require(token.transferFrom(msg.sender,cruize_wallet, amountToCruize));
+    }
+
+    function depositToAave(address reserve, uint256 amount)
+        internal
+        returns (uint256)
+    {
+        if ((amount > 0 && msg.value != 0) || (msg.value > 0 && amount != 0))
```

Comment:
this condition  must be the same for the eth and another token not seems a good practise because amount represented  the number of token